### ISSUE TITLE
Propose upstream patches from our accessibility changes

### ISSUE THREAD

**Issue by valdeza** [gbif/hp-idigbio#9]
> **Disclaimer:** I, the issue authour, am not an attorney. Any material shared in this GitHub issue is provided for informational purposes only, should not be construed as legal advice on any subject matter, is not to be acted on as such, and may be neither current nor accurate.

### Summary
Our Hosted Portal, which originally used GBIF's Jekyll theme as-is, now overrides many files in order to conform to Web Content Accessibility Guidelines (WCAG) 2.1 Level AA specification.  
We should probably share these upstream so that we do not need all of these file patches/overrides.

#### Patch summary
- Added accessible text to various site elements missing it
- Accessible colour contrast (without any drastic change to GBIF theming, in my opinion)
- Added accessibility attributes to GBIF search interface.  
  We accomplished this by using a `MutationObserver` to trail behind GBIF's `render()` call and insert attributes based on element placement and context.  
  Modifying the source code used to generate the GBIF library would be preferable to have accessibility attributes present from the start.

### Improvements
- Improved site navigation experience for all assistive technology (AT) users on any Hosted Portal  
  (presumably; I personally do not have much AT experience.)  
  ((Does the GBIF.org website use the same library for ___its___ interfaces? If so, these changes might have more impact that I initially thought.))
- If our changes get accepted, we can safely remove our Hosted Portal file overrides. Future Hosted Portal theme updates from GBIF can then be obtained with little issue.
- WCAG 2.1 Level AA conformance may be of interest to:
  - U.S. entities (such as ourselves) that must comply with the Americans with Disabilities Act and/or Section 504 of the Rehabilitation Act of 1973
  - European entities that may need to comply with the European Accessibility Act or Web Accessibility Directive (EN 301 549)
  - Any other entities with similar mandates

### Risks
- Not merging these changes into GBIF can cause unexpected site rendering issues if our then-outdated file patch is omitting necessary code that other site components are now expecting.
- WCAG 2.1 Level AA conformance alone does not automatically make a website "accessible" (in a practical and legal sense).  
  (But it's a step in a good direction that some might appreciate.)

### Possible areas of interest
- https://github.com/gbif/gbif-web/tree/main/packages/gbif-org  
  Appears to be the source code for the Javascript library that renders the GBIF search interfaces within Hosted Portals
- https://github.com/gbif/jekyll-hp-base-theme/
